admin/config-ui/class-configuration-options-adapter.php

```diff
@@ -0,0 +1,232 @@
+		}
+
+		$test = WPSEO_Options::get_option( $option );
+		if ( is_null( $test ) ) {
+			throw new InvalidArgumentException( 'Yoast option ' . $option . ' not found.' );
+		}
+
+		$this->add_lookup( $class_name, self::OPTION_TYPE_YOAST, array(
+			$option,
+			$key,
+		) );
+	}
+
+	/**
+	 * Add a lookup for a custom implementation
+	 *
+	 * @param string   $class_name   Class to bind to the lookup.
+	 * @param callable $callback_get Callback to retrieve data.
+	 * @param callable $callback_set Callback to save data.
+	 *
+	 * @throws InvalidArgumentException Thrown when invalid input is provided.
+	 */
+	public function add_custom_lookup( $class_name, $callback_get, $callback_set ) {
+		if ( ! class_exists( $class_name ) ) {
+			throw new InvalidArgumentException( 'Class must exist.' );
+		}
+
+		if ( ! is_callable( $callback_get ) || ! is_callable( $callback_set ) ) {
+			throw new InvalidArgumentException( 'Custom option must be callable.' );
+		}
+
+		$this->add_lookup( $class_name, self::OPTION_TYPE_CUSTOM, array(
+			$callback_get,
+			$callback_set,
+		) );
+	}
+
+	/**
+	 * Add a field lookup.
+	 *
+	 * @param string       $class_name Class to add lookup for.
+	 * @param string       $type       Type of lookup.
+	 * @param string|array $option     Implementation of the lookup.
+	 *
+	 * @throws Exception Thrown when invalid input is provided.
+	 */
+	protected function add_lookup( $class_name, $type, $option ) {
+		$this->lookup[ $class_name ] = array(
+			'type'   => $type,
+			'option' => $option,
+		);
+	}
+
+	/**
+	 * Get the data for the provided field
+	 *
+	 * @param WPSEO_Config_Field $field Field to get data for.
+	 *
+	 * @return mixed
+	 */
+	public function get( WPSEO_Config_Field $field ) {
+		$class_name = get_class( $field );
+
+		// Lookup option and retrieve value.
+		$type   = $this->get_option_type( $class_name );
+		$option = $this->get_option( $class_name );
+
+		switch ( $type ) {
+			case self::OPTION_TYPE_WORDPRESS:
+				return get_option( $option );
+
+			case self::OPTION_TYPE_YOAST:
+				$group = WPSEO_Options::get_option( $option[0] );
+
+				return $group[ $option[1] ];
+
+			case self::OPTION_TYPE_CUSTOM:
+				return call_user_func( $option[0] );
+		}
+
+		return null;
+	}
+
+	/**
+	 * Save data from a field
+	 *
+	 * @param WPSEO_Config_Field $field Field to use for lookup.
+	 * @param mixed              $value Value to save to the lookup of the field.
+	 *
+	 * @return bool
+	 */
+	public function set( WPSEO_Config_Field $field, $value ) {
+		$class_name = get_class( $field );
+
+		// Lookup option and retrieve value.
+		$type   = $this->get_option_type( $class_name );
+		$option = $this->get_option( $class_name );
+
+		switch ( $type ) {
+			case self::OPTION_TYPE_WORDPRESS:
+				return update_option( $option, $value );
+
+			case self::OPTION_TYPE_YOAST:
+				$group = WPSEO_Options::get_option( $option[0] );
+
+				$before = $group[ $option[1] ];
+				if ( $before === $value ) {
```

Comment:
`update_option` does the exact same check, so we don't need to: https://core.trac.wordpress.org/browser/tags/4.6/src/wp-includes/option.php#L294